### PR TITLE
Add support for schema-less HTTP proxy URLs (#4505)

### LINKED ISSUE #4505: Using proxy through http fails (https works)
### Description

When I scrape without proxy, both https and http urls work.
Using proxy through https works just fine. My problem is when I try http urls.
In that moment I get the `twisted.web.error.SchemeNotSupported: Unsupported scheme: b''` error

As I see, most of the people have this issue the other way around.

### Steps to Reproduce

1. Scrape a http link with proxy

**Expected behavior:** Get a 200 with the desired data.

**Actual behavior:**
```
ERROR: Error downloading <GET http://*********>
Traceback (most recent call last):
  File "/srv/scraper/venv/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/srv/scraper/venv/lib/python3.7/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/srv/scraper/venv/lib/python3.7/site-packages/scrapy/core/downloader/middleware.py", line 42, in process_request
    defer.returnValue((yield download_func(request=request, spider=spider)))
  File "/srv/scraper/venv/lib/python3.7/site-packages/scrapy/utils/defer.py", line 55, in mustbe_deferred
    result = f(*args, **kw)
  File "/srv/scraper/venv/lib/python3.7/site-packages/scrapy/core/downloader/handlers/__init__.py", line 76, in download_request
    return handler.download_request(request, spider)
  File "/srv/scraper/venv/lib/python3.7/site-packages/scrapy/core/downloader/handlers/http11.py", line 82, in download_request
    return agent.download_request(request)
  File "/srv/scraper/venv/lib/python3.7/site-packages/scrapy/core/downloader/handlers/http11.py", line 361, in download_request
    d = agent.request(method, to_bytes(url, encoding='ascii'), headers, bodyproducer)
  File "/srv/scraper/venv/lib/python3.7/site-packages/scrapy/core/downloader/handlers/http11.py", line 262, in request
    endpoint=self._getEndpoint(self._proxyURI),
  File "/srv/scraper/venv/lib/python3.7/site-packages/twisted/web/client.py", line 1729, in _getEndpoint
    return self._endpointFactory.endpointForURI(uri)
  File "/srv/scraper/venv/lib/python3.7/site-packages/twisted/web/client.py", line 1607, in endpointForURI
    raise SchemeNotSupported("Unsupported scheme: %r" % (uri.scheme,))
twisted.web.error.SchemeNotSupported: Unsupported scheme: b''
```

**Reproduces how often:** Every time I scrape with proxy

### Versions
```
Scrapy       : 2.0.1
lxml         : 4.4.1.0
libxml2      : 2.9.9
cssselect    : 1.1.0
parsel       : 1.5.2
w3lib        : 1.21.0
Twisted      : 20.3.0
Python       : 3.7.3 (default, Apr  3 2019, 05:39:12) - [GCC 8.3.0]
pyOpenSSL    : 19.0.0 (OpenSSL 1.1.1c  28 May 2019)
cryptography : 2.7
Platform     : Linux-4.19.0-5-amd64-x86_64-with-debian-10.0
```
### Additional context

I tried to add some breakpoints at the end to see where it cracks.
I added the following lines in "twisted/web/client/py", before the cracking point:
```python
        endpoint = HostnameEndpoint(self._reactor, host, uri.port, **kwargs)
        import logging
        logger = logging.getLogger(__name__)
        logger.error("%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%")
        logger.error(uri)
        logger.error(uri.host)
        logger.error(uri.port)
        logger.error(uri.scheme)
        logger.error(dir(uri))
        if uri.scheme == b'http':
            return endpoint
        elif uri.scheme == b'https':
            connectionCreator = self._policyForHTTPS.creatorForNetloc(uri.host,
                                                                      uri.port)
            return wrapClientTLS(connectionCreator, endpoint)
        else:
            raise SchemeNotSupported("Unsupported scheme: %r" % (uri.scheme,))
```

Apparently in this point there is no schema. If I run the same code with a https url, this code is never reached. It seems that getting up to point there is bad and the proxy is not used

(edited to apply formatting)

### PR DESCRIPTION
**Pull request by HausCloud**
Constructive criticism would be much appreciated! I manually set http using setattr on self._proxyURI and that worked. I looked at Gallaecio's suggestion and implented his solution instead.

Fixes #4505

### CODE REVIEW

**Comment by HausCloud** [comment]
Hi Gallaecio, thanks for the tip. I moved it to the init method and removed the unnecessary back and forth with the parsing. I squashed it into one commit to avoid littering the git history and didn't think the if is necessary. If you need any extra changes just let me know. I'm honestly learning a lot in all of this so thanks! P.S. I may have dropped an extra line in there :/

File: scrapy/core/downloader/handlers/http11.py [hunk 1]
```diff
@@ -250,14 +250,15 @@ def __init__(self, reactor, proxyURI, connectTimeout=None, bindAddress=None, poo
             bindAddress=bindAddress,
             pool=pool,
         )
-        self._proxyURI = URI.fromBytes(proxyURI)
+        self._proxyURI = URI.fromBytes(urlparse(proxyURI)._replace(scheme=bytes('http', 'utf-8')).geturl())
```
Comment (Gallaecio) [inline scrapy/core/downloader/handlers/http11.py:253]:
```suggestion
        self._proxyURI = URI.fromBytes(urlparse(proxyURI)._replace(scheme=b'http').geturl())
```

**Comment by Gallaecio** [comment]
The next step would be a test. I’ve had a look, and it looks like there is some existing test that you could use as a basis: `HttpProxyTestCase.test_download_with_proxy` at `tests/test_downloader_handlers.py`. The same test with a `http_proxy` without protocol should do the job, I think.

It would be great if you also locate a similar test for HTTPS, and add a similar counterpart without protocol. That way we make sure we do not break the similar feature of HTTPS in the future.

**Comment by HausCloud** [comment]
Thanks for hint in the right direction Gallaecio. I should have it done by the end of today.

**Comment by Gallaecio** [comment]
@HausCloud Did you close the pull request accidentally?

**Comment by HausCloud** [comment]
Yes I did by mistakes. I was trying to incorporate changes from scrapy into my fork. Also, I'm getting an error when I try https. I adjusted the if condition so it always handles it as https and didn't provide a schema. Below is the error I got.
Screen Shot 2020-05-06 at 4 59 55 PM

**Comment by Gallaecio** [comment]
Feel free to reopen the pull request just with the HTTP test.

**Pull request closed without merging**